RevenueCatUI/Helpers/FileImageLoader.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  FileImageLoader.swift
+//
+//
+//  Created by RevenueCat on 11/4/25.
+//
+
+import Foundation
+import RevenueCat
+
+#if canImport(UIKit)
+import UIKit
+#elseif canImport(AppKit)
+import AppKit
+#endif
+
+import SwiftUI
+@_spi(Internal) import RevenueCat
+
+@MainActor
+@available(iOS 15.0, macOS 12.0, watchOS 8.0, tvOS 15.0, *)
+final class FileImageLoader: ObservableObject {
+
+    let fileRepository: FileRepository
+    let url: URL?
+
+    init(fileRepository: FileRepository, url: URL?) {
+        self.fileRepository = fileRepository
+        self.url = url
+
+        self.loadFromCache(url: url)
+    }
+
+    typealias Value = (image: Image, size: CGSize)
+
+    private func loadFromCache(url: URL?) {
+        guard let url = url else {
+            return
+        }
+
+        self.result = fileRepository.getCachedFileURL(
+            for: url,
+            withChecksum: nil
+        )?.asImageAndSize
+
+        self.wasLoadedFromCache = self.result != nil
+    }
+
+    @Published
+    private(set) var result: Value?
+
+    private(set) var wasLoadedFromCache: Bool = false
+
+    func load() async {
+        if self.result != nil {
+            return
+        }
+
+        guard let url = self.url else {
+            return
+        }
+
+
+        Task.detached(priority: .utility) {
```

Comment:
Why launch a detached task?

Reply:
We need the `Image` creation to happen off the main thread because it was causing deadlocks thingies 🤷‍♂️

Reply:
My understanding of Tasks are that they do not guarantee that they will not be the main thread, only that they will execute on the first available thread. And Detached tasks are simply tasks that live outside of the calling context and therefore will outlive the parent context if that context is canceled. I don't think that the detached task should be needed here to get it to run off the main thread in most cases.

Reply:
In fact… we shouldn't need a task here at all, since the function itself is asynchronous

Reply:
I just ran this test 10k times and it passes every time:

```swift
    @Test()
    func isAThread() async {
        await doesThisWork()

        await Task {
            assert()
        }.value

        await Task.detached {
            assert()
        }.value

        await Task { @MainActor in
            await doesThisWork() // hops off of main
            assert(result: true)
        }.value
    }

    func doesThisWork() async {
        assert()
    }

    func assert(result: Bool = false, _ location: SourceLocation = #_sourceLocation) {
        #expect(Thread.isMainThread == result, sourceLocation: location)
    }

```

Reply:
I took out the `Task` and it it gets stuck loading the images 🤷‍♂️ It always works with it in 🙃 

<img width="433" height="966" alt="Image" src="https://github.com/user-attachments/assets/c14866cc-f223-46c8-bb36-57d2d74e5c20" />

<img width="696" height="836" alt="Image" src="https://github.com/user-attachments/assets/51ca280e-14d5-453f-80b6-9f7d1058f11e" />